sklearn/utils/testing.py

```diff
@@ -302,24 +307,28 @@ def ignore_warnings(obj=None):
     42
 
     """
+
+    def _ignore_warnings(fn):
+        """Decorator to catch and hide warnings without visual nesting"""
+        @wraps(fn)
+        def wrapper(*args, **kwargs):
+            # very important to avoid uncontrolled state propagation
+            clean_warning_registry()
+            with warnings.catch_warnings():
+                if category is None:
+                    warnings.simplefilter("ignore")
+                else:
+                    warnings.simplefilter("ignore", category)
+                return fn(*args, **kwargs)
+
+        return wrapper
+
     if callable(obj):
         return _ignore_warnings(obj)
-    else:
+    elif category is None:
         return _IgnoreWarnings()
```

Comment:
What I had in mind was to extend `_IgnoreWarnings` to accept a `cagegory` keyword that it would itself pass to its own underlying call to `warnings.simplefilter` in its `__enter__` method.